origins_common/origins_common.module

```diff
@@ -85,3 +85,59 @@ function origins_common_form_entity_embed_dialog_alter(&$form, FormStateInterfac
   $form['attributes']['data-align']['#default_value'] = 'right';
   $form['attributes']['data-align']['#access'] = FALSE;
 }
+
+/**
+ * Implements hook_preprocess_table().
+ */
+function origins_taxonomy_access_preprocess_table(&$variables) {
+  // Add new 'Revision Number' column to the node revisions tab.
+  //
+  // Making this change in preprocess is not beautiful, but seems
+  // to be the only workable solution as the 'diff' module has
+  // already been enabled and has taken over the core route with
+  // its own controller. If we created our own controller to do
+  // this then we would have to call the contrib controller,
+  // which does not sound like a good idea.
+  if (\Drupal::routeMatch()->getRouteName() == 'entity.node.version_history') {
+    // Add 'Revision Number' column header.
+    $newhead = ['tag' => 'th', 'content' => ['#markup' => t('Revision Number')]];
+    // Insert this new column after the first one.
+    // Use array_splice as we don't know exactly how many columns there will be
+    // (and this code should work whether the 'diff' module has been enabled or
+    // not).
+    array_splice($variables['header'], 1, 0, [$newhead]);
+    if (isset($variables['header']['select_column_one'])) {
+      $variables['header']['select_column_one']['content'] = ['#markup' => t('Compare 1')];
```

Comment:
Haven't seen this in-situ yet, but is there a more descriptive label than 'Compare 1' and 'Compare 2'? What does the 1 and 2 refer to?

Reply:
Well, the 'diff' module adds a 'Compare selected revisions' button at the bottom of the form and then just adds two columns of radios. No headings are added to the columns but the user is obviously expected to click one in each column and then hit the compare button. I thought that something was needed (especially after I added a new column and made the page more busy) so I came up with 'Compare 1' and 'Compare 2' - I'm not delighted with them so am open to suggestions !

Reply:
Have now gone for 'Compare revision 1' and 'Compare revision 2' as agreed